File: js-pr/56/javascript/code-quality-error-handling/calculator_29.js
// {fact rule=code-quality-error-handling@v1.0 defects=0}
function divide(a, b) {
    try {
        if (b === 0) {
            throw new Error('Division by zero');
        }
        return a / b;
    } catch (error) {
        console.error('Calculation error:', error.message);
        return null;
    }
}
// {/fact}

File: python-pr/52/python/code-quality-error-handling/calculator_29.py
# {fact rule=code-quality-error-handling@v1.0 defects=0}
def divide(a, b):
    try:
        if b == 0:
            raise ValueError("Division by zero")
        return a / b
    except ValueError as e:
        print(f"Calculation error: {e}")
        return None
# {/fact}